src/Stack/Fetch.hs

```diff
@@ -554,6 +533,69 @@ fetchPackages' mdistDir toFetchAll = do
                 S.writeFile cabalFP $ tfCabal toFetch
 
                 atomically $ modifyTVar outputVar $ Map.insert ident destDir
+            $logWarn $ mconcat $ map (\(path, entryType) -> "Unexpected entry type " <> entryType <> " for entry " <> T.pack path) unexpectedEntries
+
+-- | Internal function used to unpack tarball.
+--
+-- Takes a path to a .tar.gz file, the name of the directory it should contain,
+-- and a destination folder to extract the tarball into. Returns unexpected
+-- entries, as pairs of paths and descriptions.
+untar :: FilePath -> FilePath -> FilePath -> IO [(FilePath, T.Text)]
```

Comment:
> but this is not trivial since this code has to talk to non-path-ified code.

Yes, I guess for the return value that is only going to be printed it doesn't matter as much. But for the input types it would be great, so we can encourage further usage of `Path` in any callers of `untar`.

> Also, the existing code uses FilePath quite a lot, see all the FP.</>.

Yes, there's a lot of room for improvement in this codebase! ;)